.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+concurrency:
+  # Group concurrency on workflow, then:
+  # - Is merge run? Group `push` events
+  # - Is pull request? Group `pull_request` events
+  group: >-
```

Comment:
I will add more comments to explain. It's the GitHub Actions event properties that are not intuituve because they are different between `push` and `pull_request` events (triggers). The way it should work is:
- Only 1 active workflow run on the `main` branch, consecutive workflow runs on `main` are queued
- Only 1 active workflow run per pull request **branch**, latest run cancels active runs

Reply:
I made a false assumption based on GitHub Docs. I will file a complaint immediately 😄

There is no `github.base_ref` property for `push` events despite what's listed in the example event in the docs.

Changing the key for merge to main (`push` events) to `github.ref` (`refs/heads/main`)